jdbc-v2/src/main/java/com/clickhouse/jdbc/internal/JdbcConfiguration.java

```diff
@@ -55,77 +39,206 @@ public boolean isIgnoreUnsupportedRequests() {
         return isIgnoreUnsupportedRequests;
     }
 
-    public JdbcConfiguration(String url, Properties info) {
-        this.allProperties = new ConcurrentHashMap<>();
-        info.forEach((k, v) -> allProperties.put(k.toString(), v.toString()));
-        this.jdbcUrl = url;//Raw URL
-        this.url = cleanUrl(url);
-        this.user = info.getProperty("user", "default");
-        this.password = info.getProperty("password", "");
+    /**
+     * Parses URL to get property and target host.
+     * Properties that are passed in the {@code info} parameter will override that are set in the {@code url}.
+     * @param url - JDBC url
+     * @param info - Driver and Client properties.
+     */
+    public JdbcConfiguration(String url, Properties info) throws SQLException {
         this.disableFrameworkDetection = Boolean.parseBoolean(info.getProperty("disable_frameworks_detection", "false"));
+        this.clientProperties = new HashMap<>();
+        this.driverProperties = new HashMap<>();
+
+        Map<String, String> urlProperties = parseUrl(url);
+        String tmpConnectionUrl = urlProperties.remove(PARSE_URL_CONN_URL_PROP);
+        initProperties(urlProperties, info);
+
+        // after initializing all properties - set final connection URL
+        boolean useSSL = Boolean.parseBoolean(info.getProperty("ssl", "false"));
+        this.connectionUrl = createConnectionURL(tmpConnectionUrl, useSSL);
         this.isIgnoreUnsupportedRequests= Boolean.parseBoolean(getDriverProperty(DriverProperties.IGNORE_UNSUPPORTED_VALUES.getKey(), "false"));
     }
 
     public static boolean acceptsURL(String url) {
+        // TODO: should be also checked for http/https
         return url.startsWith(PREFIX_CLICKHOUSE) || url.startsWith(PREFIX_CLICKHOUSE_SHORT);
     }
 
-    protected String cleanUrl(String url) {
-        url = stripUrlPrefix(url);
-        boolean setSSL = false;
-        boolean ssl = false;
-        try {
-            ssl = Boolean.parseBoolean(allProperties.get("ssl"));
-            setSSL = true;
-        } catch (Exception e) {
-            log.trace("Failed to parse SSL property.", e);
-        }
+    public String getConnectionUrl() {
+        return connectionUrl;
+    }
 
+    /**
+     * Returns normalized URL that can be passed as parameter to Client#addEndpoint().
+     * Returned url has only schema and authority and doesn't have query parameters or path.
+     * JDBC URL should have only a single path parameter to specify database name.
+     * Note: Some BI tools do not let pass JDBC URL, so ssl is passed as property.
+     * @param url - JDBC url
+     * @param ssl - if SSL protocol should be used when protocol is not specified
+     * @return URL without JDBC prefix
+     */
+    static String createConnectionURL(String url, boolean ssl) throws SQLException {
         if (url.startsWith("//")) {
-            if (setSSL) {
```

Comment:
Maybe i'am missing it. This we did for logic if we have properties like in tableau? are we covering it

Reply:
yes, 
this method is simply builds string. `ssl` is now external parameter - it will be set to the value of `ssl` property once all properties from user and url are merged.

Reply:
I have a test for ssl connection and have tests where ssl is passed as property.

Reply:
Great thanks for the clarification